web_src/js/components/RepoActionView.vue

```diff
@@ -188,11 +188,22 @@ const sfc = {
     },
 
     appendLogs(stepIndex, logLines, startTime) {
+      // position of the client view relative to the website top
+      const clientHeight = document.documentElement.clientHeight + window.scrollY;
+      // height of the logs container relative to the website top
+      const logsContainerHeight = this.$refs.stepsContainer.getBoundingClientRect().bottom + window.scrollY;
+
       for (const line of logLines) {
         // TODO: group support: ##[group]GroupTitle , ##[endgroup]
         const el = this.getLogsContainer(stepIndex);
         el.append(this.createLogLine(line, startTime, stepIndex));
       }
+
+      // scrolls to the bottom if job is running and the bottom of the logs container is visible
+      if (!this.run.done && logLines.length > 0 && clientHeight >= logsContainerHeight) {
+        const newLogsContainerHeight = this.$refs.stepsContainer.getBoundingClientRect().bottom + window.scrollY;
+        window.scrollTo({top: clientHeight + (newLogsContainerHeight - logsContainerHeight), behavior: 'smooth'});
```

Comment:
Does it have to be `smooth`? Personally I hate all kinds of smooth scrolling. We could set `auto` here which would then enable themes to change the behaviour via CSS [`scroll-behaviour`](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior) if they so desire.

Reply:
With smooth scrolling it's clearer to see when new lines were added, especially when the newly added lines are looking similar. I often have action logs on my second screen to see if the changes I've made are passing while I continue working on my main screen, and with smooth scrolling, I notice new logs faster.

<details>
<summary>Videos of both scroll behaviors</summary>

https://github.com/go-gitea/gitea/assets/63594396/ed9acecd-7bc3-44a9-a141-c4f614837642

https://github.com/go-gitea/gitea/assets/63594396/7e61b5b8-e82b-48d1-a1f3-142d4e9a3460
</details>

Reply:
I guess it's an argument for it, yes.